Modules/TestQuestionPool/classes/class.assMatchingQuestion.php

```diff
@@ -334,11 +339,11 @@ public function loadFromDb($question_id): void
             array('integer'),
             array($question_id)
         );
-        include_once "./Modules/TestQuestionPool/classes/class.assAnswerMatchingDefinition.php";
+
         $this->definitions = array();
         if ($result->numRows() > 0) {
             while ($data = $ilDB->fetchAssoc($result)) {
-                $definition = new assAnswerMatchingDefinition($data['definition'], $data['picture'], $data['ident']);
+                $definition = $this->createMatchingDefinition($data['definition'], $data['picture'], (int)$data['ident']);
```

Comment:
```suggestion
                $definition = $this->createMatchingDefinition($data['definition'] ?? '', $data['picture'] ?? '', (int) $data['ident']);
```
This can now fail, if $data['definition'] === null or $data['picture'] === null